homestar-runtime/src/settings.rs

```diff
@@ -22,6 +30,115 @@ pub struct Monitoring {
     process_collector_interval: u64,
 }
 
+#[derive(Clone, Debug, Deserialize)]
+/// Configure how the Network keypair is generated or using an existing one
+pub(crate) enum PubkeyConfig {
+    #[serde(rename = "random")]
+    Random,
+    /// Seed string should be a base64 encoded 32 bytes. This is used as the RNG seed to generate a ed25519 key.
+    #[serde(rename = "random_seed")]
+    GenerateFromSeed(PupkeyRNGSeed),
+    /// File path to a PEM encoded ed25519 key
+    #[serde(rename = "existing")]
+    Existing(ExistingKeyPath),
+}
+
+/// Supported key types of homestar
+#[derive(Clone, Debug, Default, Deserialize)]
+pub(crate) enum KeyType {
+    #[default]
+    #[serde(rename = "ed25519")]
+    Ed25519,
+    #[serde(rename = "secp256k1")]
+    Secp256k1,
+}
+
+/// Seed material for RNG generated keys
+#[serde_as]
+#[derive(Clone, Debug, Deserialize)]
+pub(crate) struct PupkeyRNGSeed {
+    #[serde(default)]
+    key_type: KeyType,
+    #[serde_as(as = "Base64")]
+    seed: [u8; 32],
+}
+
+/// Info on where and what the Key file is
+#[derive(Clone, Debug, Deserialize)]
+pub(crate) struct ExistingKeyPath {
+    #[serde(default)]
+    key_type: KeyType,
+    path: String,
+}
+
+impl PubkeyConfig {
+    /// Produce a Keypair using the given configuration.
+    pub(crate) fn generate_keypair(&self) -> anyhow::Result<identity::Keypair> {
+        match self {
+            PubkeyConfig::Random => Ok(identity::Keypair::generate_ed25519()),
+            PubkeyConfig::GenerateFromSeed(PupkeyRNGSeed { key_type, seed }) => {
+                // seed RNG with supplied seed
+                let mut r = rand::prelude::StdRng::from_seed(*seed);
+                let mut new_key: [u8; 32] = r.gen();
+
+                match key_type {
+                    KeyType::Ed25519 => {
+                        identity::Keypair::ed25519_from_bytes(new_key).map_err(|e| {
+                            anyhow!("Failed to generate ed25519 key from random: {:?}", e)
```

Comment:
minor: `{:#?}` instead of `{:?}` will pretty-print debug logs.

Reply:
yeah we can pretty print these. though if we are going to be showing these errors to a lib that displays errors better in general like https://docs.rs/color-eyre/latest/color_eyre/

Reply:
@mriise yep, we'll prob bring in color_eyre for local dev. Nonetheless, this is more so for logs that will be picked up by grafana/other-simialr tool (or logging service), especially as we use logfmt.